### ISSUE TITLE
Crash when call not in UI Thread

### ISSUE THREAD

**Issue by snechiporenko** [jraska/Falcon#29]
Sometimes there is a fall. 
I think it is related to the display of dialogs

```
Fatal Exception: java.lang.NullPointerException: Attempt to read from field 'android.graphics.Rect android.view.View$AttachInfo.mTmpInvalRect' on a null object reference
       at android.view.View.onDrawScrollIndicators(View.java:14742)
       at android.view.View.onDrawForeground(View.java:18706)
       at android.view.View.draw(View.java:17200)
       at android.widget.AbsListView.draw(AbsListView.java:4322)
       at android.view.View.draw(View.java:17097)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
       at android.view.View.draw(View.java:17095)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
       at android.view.View.draw(View.java:17095)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
       at android.view.View.draw(View.java:17095)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
       at android.view.View.draw(View.java:17095)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
       at android.view.View.draw(View.java:17095)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
       at android.view.View.draw(View.java:17095)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
       at android.view.View.draw(View.java:17192)
       at com.android.internal.policy.DecorView.draw(DecorView.java:756)
       -> at com.jraska.falcon.Falcon.drawRootToBitmap(Falcon.java:162)
       -> at com.jraska.falcon.Falcon.drawRootsToBitmap(Falcon.java:147)
       -> at com.jraska.falcon.Falcon.access$300(Falcon.java:29)
       -> at com.jraska.falcon.Falcon$1.run(Falcon.java:132)**
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6186)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
```
Maybe you can add catch to this block:
```
      activity.runOnUiThread(new Runnable() {
        @Override
        public void run() {
          try {
            drawRootsToBitmap(viewRoots, bitmap);
          } finally {
            latch.countDown();
          }
        }
      });
```
Thanks for your work.

**Comment by jraska** [comment]
Hello, thank you for reporting this. Does it happen always for your case or only sometimes?

I think I have a fix in my head and I will do it next week.

**Comment by jraska** [comment]
Fix made and released version `2.0.1`

Right now the app does not crash, but we still throw `UnableToTakeScreenshotException` to be consistent with rest of the library. Anyway, we do it on correct thread calling `Falcon.takeScreenshot` not on main thread, so we either fail the test only, or one can catch this exception.

More about why we use this exception and not only logging can be found here: https://github.com/jraska/Falcon/issues/19
